packages/expo-updates/android/src/androidTest/java/expo/modules/updates/launcher/SelectionPolicyFilterAwareTest.java

```diff
@@ -0,0 +1,147 @@
+    updateRollout1 = NewManifest.fromManifestJson(manifestJsonRollout1, config).getUpdateEntity();
+
+    JSONObject manifestJsonDefault2 = new JSONObject("{\"id\":\"079cde35-8433-4c17-81c8-7117c1513e74\",\"createdAt\":\"2021-01-13T19:39:22.480Z\",\"runtimeVersion\":\"1.0\",\"launchAsset\":{\"hash\":\"DW5MBgKq155wnX8rCP1lnsW6BsTbfKLXxGXRQx1RcOA\",\"key\":\"0436e5821bff7b95a84c21f22a43cb96.bundle\",\"contentType\":\"application/javascript\",\"url\":\"https://url.to/bundle\"},\"assets\":[{\"hash\":\"JSeRsPNKzhVdHP1OEsDVsLH500Zfe4j1O7xWfa14oBo\",\"key\":\"3261e570d51777be1e99116562280926.png\",\"contentType\":\"image/png\",\"url\":\"https://url.to/asset\"}],\"updateMetadata\":{\"releaseName\":\"default\"}}");
+    updateDefault2 = NewManifest.fromManifestJson(manifestJsonDefault2, config).getUpdateEntity();
+
+    JSONObject manifestJsonRollout2 = new JSONObject("{\"id\":\"079cde35-8433-4c17-81c8-7117c1513e75\",\"createdAt\":\"2021-01-14T19:39:22.480Z\",\"runtimeVersion\":\"1.0\",\"launchAsset\":{\"hash\":\"DW5MBgKq155wnX8rCP1lnsW6BsTbfKLXxGXRQx1RcOA\",\"key\":\"0436e5821bff7b95a84c21f22a43cb96.bundle\",\"contentType\":\"application/javascript\",\"url\":\"https://url.to/bundle\"},\"assets\":[{\"hash\":\"JSeRsPNKzhVdHP1OEsDVsLH500Zfe4j1O7xWfa14oBo\",\"key\":\"3261e570d51777be1e99116562280926.png\",\"contentType\":\"image/png\",\"url\":\"https://url.to/asset\"}],\"updateMetadata\":{\"releaseName\":\"rollout\"}}");
+    updateRollout2 = NewManifest.fromManifestJson(manifestJsonRollout2, config).getUpdateEntity();
+
+    JSONObject manifestJsonNested = new JSONObject("{\"id\":\"079cde35-8433-4c17-81c8-7117c1513e72\",\"createdAt\":\"2021-01-11T19:39:22.480Z\",\"runtimeVersion\":\"1.0\",\"launchAsset\":{\"hash\":\"DW5MBgKq155wnX8rCP1lnsW6BsTbfKLXxGXRQx1RcOA\",\"key\":\"0436e5821bff7b95a84c21f22a43cb96.bundle\",\"contentType\":\"application/javascript\",\"url\":\"https://url.to/bundle\"},\"assets\":[{\"hash\":\"JSeRsPNKzhVdHP1OEsDVsLH500Zfe4j1O7xWfa14oBo\",\"key\":\"3261e570d51777be1e99116562280926.png\",\"contentType\":\"image/png\",\"url\":\"https://url.to/asset\"}],\"updateMetadata\":{\"nested\":{\"object\":{\"key\": \"value\"}}}}");
+    updateNested = NewManifest.fromManifestJson(manifestJsonNested, config).getUpdateEntity();
+  }
+
+  @Test
+  public void testSelectUpdateToLaunch() {
+    // should pick the newest update that matches the manifest filters
+    UpdateEntity expected = updateRollout1;
+    UpdateEntity actual = selectionPolicy.selectUpdateToLaunch(Arrays.asList(updateDefault1, expected, updateDefault2), manifestFilters);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_OlderMatching() {
```

Comment:
It may make sense to change this test to "SecondNewestMatching" and to add updateRollout3 to show that matching manifests still are deleted if they aren't the newest or second newest.